examples/cvae/cvae.py

```diff
@@ -0,0 +1,188 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+import numpy as np
+from pathlib import Path
+import pyro
+import pyro.distributions as dist
+from pyro.infer import SVI, Trace_ELBO
+import torch
+import torch.nn as nn
+from tqdm import tqdm
+
+
+class Encoder(nn.Module):
+    def __init__(self, z_dim, hidden_1, hidden_2):
+        super().__init__()
+        self.fc1 = nn.Linear(784, hidden_1)
+        self.fc2 = nn.Linear(hidden_1, hidden_2)
+        self.fc31 = nn.Linear(hidden_2, z_dim)
+        self.fc32 = nn.Linear(hidden_2, z_dim)
+        self.relu = nn.ReLU()
+
+    def forward(self, x, y):
+        # put x and y together in the same image for simplification
```

Comment:
what exactly is going on here?

Reply:
Let's say we are training the model using 1 quadrant of a MNIST digit as input X (bottom left), which means the remaining quadrants (top left, top right, bottom right) are the target output Y to be predicted. This means:
- In the input X, all pixels that are not in the bottom left quadrant are set to -1
- In the output Y, all pixels that are not in the top left, top right and bottom right quadrants (i.e. the pixels in the bottom left quadrant) are set to -1
- So X, and Y are complementary: at training time, they form perfectly an original MNIST digit, while at inference time, they form an approximate MNIST digit (at inference time, Y is not provided: as explained in the paper, at inference time we use the prior net to approximate Y_hat, using it instead of Y)

Lines 25 and 26 just put X and Y (or Y_hat) together in a single 28 x 28 image, to be fed either into the prior net or into the recognition net. 

There's another way to interpret & implement the paper: Instead of putting X and Y (or Y_hat) together as I did, I could have:
- Removed all pixels marked with -1 from X and Y (or Y_hat)
- Concatenated the flattened tensors and carried on

First choice seemed simpler & more elegant to me... But I can do the 2nd way as well, no problem :) (need to test it, but in theory it should work as well as the first approach)